api/v1beta1/infraenv_types.go

```diff
@@ -103,6 +103,11 @@ type InfraEnvSpec struct {
 	// Applicable for both iPXE, and ISO streaming from Image Service.
 	// +optional
 	KernelArguments []KernelArgument `json:"kernelArguments,omitempty"`
+
+	// IronicAgentImageOverride is the pull spec for the ironic agent image to use when discovering hosts
```

Comment:
It sounds like the core use case for end users is when they want to install a spoke using BareMetalHosts for hardware that's a different CPU architecture than the hub. I think that, or some similar guidance and context, is worth documenting here. When someone reads the API docs, they should understand when/why this feature is useful and applicable.

Reply:
We know have a CpuArchitecture property in the InfraEnv - could we use that to infer the correct image?

Reply:
We [currently](https://github.com/openshift/assisted-service/blob/8fbbce6515dc38ad028e5a47417c601a0a0c5495/internal/controller/controllers/preprovisioningimage_controller.go#L446-L448) use that to know when the hub image is incorrect for a particular infraenv, but if the hub is using a single-arch release image then there's no way for us to determine what the pull spec is for some other architecture ironic agent.

Reply:
That makes me think again about what we expect the user to do.

They could run the following. Is it what we'd recommend?

```
$ oc adm release info quay.io/openshift-release-dev/ocp-release:4.12.0-multi --filter-by-os=linux/arm64 --image-for=ironic-agent
W0117 09:51:02.049979   80777 manifest.go:449] Chose linux/arm64 manifest from the manifest list.
quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:69c5e28d0cbe374a6a663f85420e33c50f363c5084d3c36e82cfc9099f6a41b1
```

That leads to the question: can we just do that automatically if connected? And if disconnected, could we pick a reasonable release from those that are available? (choosing from the mirrored releases in the hive CRD whose name is escaping me at the moment)

In any case, it seems like we need to give the user fairly specific guidance on how to figure out what value to put here. If we can give them specific guidance on what to do, maybe instead we can just do it?

Reply:
Sure, but the idea was to make this easy to manage for everyone involved, us included.

For something like 95% of cases the spoke cluster arch is going to match the hub and this value won't be used. When the hub release image is multi arch, devs will probably be the only ones using it.

This way we'll get the exact image we need from the hub release image, our documentation will say "mirror the hub release image if you're disconnected" and everything will just work without us having to constantly maintain some list of images for every possible hub version.